components/org.wso2.carbon.identity.conditional.auth.functions.opa/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/opa/internal/OPAFunctionsServiceComponent.java

```diff
@@ -0,0 +1,118 @@
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.opa.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.*;
+import org.wso2.carbon.identity.application.authentication.framework.JsFunctionRegistry;
+import org.wso2.carbon.identity.claim.metadata.mgt.ClaimMetadataManagementService;
+import org.wso2.carbon.identity.conditional.auth.functions.opa.InvokeOpaFunction;
+import org.wso2.carbon.identity.conditional.auth.functions.opa.InvokeOpaFunctionImpl;
+import org.wso2.carbon.identity.core.util.IdentityCoreInitializedEvent;
+
+
+/**
+ * OSGi declarative services component which handle cookie related conditional auth functions.
+ */
+
+@Component(
+        name = "identity.conditional.auth.functions.opa.component",
+        immediate = true
+)
+public class OPAFunctionsServiceComponent {
+
+    private static final Log LOG = LogFactory.getLog(OPAFunctionsServiceComponent.class);
+
+    public static final String FUNC_INVOKE_OPA = "invokeOPA";
+
+
+    @Activate
+    protected void activate(ComponentContext ctxt) {
+
+        try {
+            JsFunctionRegistry jsFunctionRegistry = OPAFunctionsServiceHolder.getInstance().getJsFunctionRegistry();
+            InvokeOpaFunction invokeOPA = new InvokeOpaFunctionImpl();
+            jsFunctionRegistry.register(JsFunctionRegistry.Subsystem.SEQUENCE_HANDLER, FUNC_INVOKE_OPA, invokeOPA);
+        } catch (Throwable e) {
+            LOG.error(e);
+        }
+
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext ctxt) {
+
+        JsFunctionRegistry jsFunctionRegistry = OPAFunctionsServiceHolder.getInstance().getJsFunctionRegistry();
+        if (jsFunctionRegistry != null) {
+            jsFunctionRegistry.deRegister(JsFunctionRegistry.Subsystem.SEQUENCE_HANDLER, FUNC_INVOKE_OPA);
+        }
+    }
+
+    @Reference(
+            service = JsFunctionRegistry.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetJsFunctionRegistry"
+    )
+    public void setJsFunctionRegistry(JsFunctionRegistry jsFunctionRegistry) {
+
+        OPAFunctionsServiceHolder.getInstance().setJsFunctionRegistry(jsFunctionRegistry);
+    }
+
+    public void unsetJsFunctionRegistry(JsFunctionRegistry jsFunctionRegistry) {
+
+        OPAFunctionsServiceHolder.getInstance().setJsFunctionRegistry(null);
+    }
+
+    @Reference(
+            name = "identityCoreInitializedEventService",
+            service = IdentityCoreInitializedEvent.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetIdentityCoreInitializedEventService")
+    protected void setIdentityCoreInitializedEventService(IdentityCoreInitializedEvent identityCoreInitializedEvent) {
```

Comment:
check whether we are using identityCoreInitializedEvent for OPA related functions. If no usage, let's remove this.